File: Problem 003/solution.c
#include <math.h>
#include <stdio.h>


long long maxPrimeFactors(long long n)
{
	long long maxPrime = -1;

	while (n % 2 == 0) {
		maxPrime = 2;
		n >>= 1; // equivalent to n /= 2
	}

	for (int i = 3; i <= sqrt(n); i += 2) {
		while (n % i == 0) {
			if (i > maxPrime)
                maxPrime = i;
			n = n / i;
		}
	}

	if (n > 2)
		maxPrime = n;

	return maxPrime;
}

int main()
{
	int t;
    scanf("%d", &t);
    for (size_t _ = 0; _ < t; _++) {
        long long n;
        scanf("%lld", &n);
        printf("%lld\n", maxPrimeFactors(n));
    }

	return 0;
}

File: Problem 003/solution.py
import math

for _ in range(int(input())):
    n = int(input())

    largest_prime = 2
    while n % 2 == 0:
        n //= 2

    for i in range(3, int(math.sqrt(n)) + 1, 2):
        while n % i == 0:
            if i > largest_prime:
                largest_prime = i
            n //= i

    if n > 2:
        largest_prime = n

    print(int(largest_prime))
